app/models/video.py

```diff
@@ -1,4 +1,28 @@
 from app import db
+from flask import current_app
+
 
 class Video(db.Model):
-    id = db.Column(db.Integer, primary_key=True)
\ No newline at end of file
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    release_date = db.Column(db.DateTime)
+    total_inventory = db.Column(db.Integer)
+
+    rental = db.relationship("Rental",passive_deletes=True, backref="videos")
+
+
+    def create_video_dict(self):
+        return_dict = {
+                "id":self.id,
+                "title":self.title,
+                "release_date":self.release_date,
+                "total_inventory": self.total_inventory
+                }
+        return return_dict
+
+    def available_inventory(self):
+        from app.models.rental import Rental
+        out_count = Rental.query.filter_by(video_id = self.id).count()
+        available_inventory = self.total_inventory - out_count
+        return available_inventory
```

Comment:
Rather than querying the database every time we need to update `available_inventory`, we can take advantage of the `rental` relationship attribute defined earlier in this class. `self.rental` will generate a list of rental instances containing a specific video (aka list of rental objects with a video_id of 2). Every time we `check-out` or `check-in` a video to/from a customer, the `self.rental` also updates to reflect the newest count of rentals for a video. 

```suggestion
        out_count = len(self.rental)
        available_inventory = self.total_inventory - out_count
        return available_inventory
```

Reply:
If y'all want to see the output of `self.rental` we can utilize print statements and the [`vars` function ](https://www.programiz.com/python-programming/methods/built-in/vars)to display the rental instance objects in dictionary form. 


```suggestion
    def available_inventory(self):
        print(self.rental)
        for video in self.rental:
            print(vars(video))
        out_count = Rental.query.filter_by(video_id = self.id).count()
        available_inventory = self.total_inventory - out_count
        return available_inventory
```